src/Axe/Managers/CompilerManager.cs

```diff
@@ -22,31 +23,47 @@ namespace Axe.Managers
     /// </summary>
     public class CompilerManager : ManagerBase, ICompilerManager
     {
-        public CompilerManager(AxeDbContext context) : base(context)
+        private INodeServices nodeServices;
+
+        public CompilerManager(AxeDbContext context, INodeServices nodeServices) : base(context)
         {
+            this.nodeServices = nodeServices;
         }
         
-        public async Task<List<CodeBlockVm>> GetCodeBlocks()
+        public async Task<List<CodeBlockVm>> GetCodeBlocks(int technologyId)
         {
-            List<CodeBlockVm> list = await context.CodeBlock.Select(cb => new CodeBlockVm
+            List<CodeBlockVm> list = await context.CodeBlock
+                .Include(cb => cb.Technology)
+                .Select(cb => new CodeBlockVm
+                {
+                    Id = cb.Id,
+                    Task = cb.Task,
+                    Technology = cb.Technology,
+                })
+                .ToListAsync();
+            
+            if (technologyId > 0)
             {
-                Id = cb.Id,
-                Task = cb.Task
-            }).ToListAsync();
+                list = list.Where(cb => cb.Technology != null && cb.Technology.Id == technologyId).ToList();
```

Comment:
first we perform .ToListAsync();
then if (technologyId > 0) we do filtering and another .ToList(); in memory
it would better if search is performed in db while quering codeblocks from context

Reply:
fe30448